Sources/Plasma/PubUtilLib/plNetGameLib/Private/plNglTrans.cpp

```diff
@@ -252,31 +259,36 @@ void NetTransCancelByConnId (unsigned connId, ENetError error) {
 //============================================================================
 void NetTransCancelAll (ENetError error) {
     hsLockGuard(s_critsect);
-    NetTrans * trans = s_transactions.Head();
-    for (; trans; trans = trans->m_link.Next())
+    for (NetTrans* trans : s_transactions) {
         CancelTrans_CS(trans, error);
+    }
 }
 
 //============================================================================
 void NetTransUpdate () {
-    LISTDECL(NetTrans, m_link) completed;
-    LISTDECL(NetTrans, m_link) parentCompleted;
+    std::list<NetTrans*> completed;
+    std::list<NetTrans*> parentCompleted;
 
     {
         hsLockGuard(s_critsect);
 
-        NetTrans * next, * trans = s_transactions.Head();
-        for (; trans; trans = next) {
-            next = s_transactions.Next(trans);
+        for (auto it = s_transactions.begin(); it != s_transactions.end();) {
+            NetTrans* trans = *it;
+            // Increment a copy of the iterator here already,
+            // because the original iterator may be invalidated by an erase call below.
+            auto next = it;
+            next++;
 
             bool done = false;
             while (!done) {
                 switch (trans->m_state) {
                     case kTransStateComplete:
-                        if (trans->m_hasSubTrans)
-                            parentCompleted.Link(trans);
-                        else
-                            completed.Link(trans);
+                        s_transactions.erase(it);
+                        if (trans->m_hasSubTrans) {
+                            parentCompleted.push_back(trans);
+                        } else {
+                            completed.push_back(trans);
+                        }
```

Comment:
I think it would be more efficient to use `std::list<T>::splice()` to avoid deleting and reallocating list nodes.